planning/autoware_mission_planner_universe/src/mission_planner/arrival_checker.cpp

```diff
@@ -55,15 +60,38 @@ bool ArrivalChecker::is_arrived(const PoseStamped & pose) const
     return false;
   }
 
-  // Check distance.
-  if (distance_ < autoware_utils::calc_distance2d(pose.pose, goal.pose)) {
+  // Compute position difference in goal frame
+  const double dx = pose.pose.position.x - goal.pose.position.x;
+  const double dy = pose.pose.position.y - goal.pose.position.y;
+
+  const double yaw_goal = tf2::getYaw(goal.pose.orientation);
+  const double longitudinal_offset_to_goal = std::cos(yaw_goal) * dx + std::sin(yaw_goal) * dy;
+  const double lateral_offset_to_goal = std::sin(yaw_goal) * dx + std::cos(yaw_goal) * dy;
```

Comment:
@Ericpotato 
```suggestion
  const double lateral_offset_to_goal = -std::sin(yaw_goal) * dx + std::cos(yaw_goal) * dy;
```
Is this missing a negative sign? 
The standard 2D rotation matrix is
$$dx' = \cos\theta \cdot dx - \sin\theta \cdot dy $$
$$dy' = \sin\theta \cdot dx + \cos\theta \cdot dy $$ 

Considering $$θ = -yaw\\_goal$$,
$$longitudinal\\_offset\\_to\\_goal = dx' = \cos(-yaw\\_goal) \cdot dx - \sin(-yaw\\_goal) \cdot dy = \cos(yaw\\_goal) \cdot dx + \sin(yaw\\_goal) \cdot dy$$
$$lateral\\_offset\\_to\\_goal = dy' = \sin(-yaw\\_goal) \cdot dx + \cos(-yaw\\_goal) \cdot dy = -\sin(yaw\\_goal) \cdot dx + \cos(yaw\\_goal) \cdot dy$$